src/main/java/com/zaxxer/nuprocess/windows/WindowsProcess.java

```diff
@@ -250,6 +250,53 @@ NuProcess start(List<String> commands, String[] environment, Path cwd)
       return this;
    }
 
+   private void attachJob()
+   {
+      if (!CHILD_TERMINATES_WITH_PARENT) {
+         return;
+      }
+
+      if (!AssignProcessToJobObject(hJob, getPidHandle())) {
+         throw new RuntimeException("Failed to attach to job " + Native.getLastError());
```

Comment:
You're almost certainly going to leak a process if this happens. `start` will catch this exception and (from my own changes) log it, and then return without the process ever being terminated.